packages/core/core/prompt/files-to-parts.ts

```diff
@@ -0,0 +1,38 @@
+import { MessageFile } from '@ai-sdk/ui-utils';
+import { ImagePart, TextPart } from './content-part';
+
+type ContentPart = TextPart | ImagePart;
+
+/**
+ * Converts a data URL of type text/* to a text string.
+ */
+export function dataUrlToText(dataUrl: string): string {
+  return atob(dataUrl.split(',')[1]);
```

Comment:
I vaguely remember some compatibility issues with `atob`

Reply:
addressed this here https://github.com/vercel/ai/pull/2226#discussion_r1677603635